### ISSUE TITLE
Can't print at all in IE11

### ISSUE THREAD

**Issue by hyperplanes** [google/material-design-lite#4129]
I know mdl isn't optimized for printing, but this seems pretty serious. Merely adding the css file to a page prevents it from printing at all, and causes print preview to freeze in IE11.

**Comment by sgomes** [comment]
@hyperplanes If we're talking about a freeze from plain CSS, it sounds like a fundamental issue with the browser engine that we may not be able to do much about, particularly if we don't know where it's happening, and whether it triggers for all users.

Have you tried a non-minified version of the MDL CSS, to see if the problem still occurs there? You can find `material.css` (the non-minified file) inside the release package.

**Comment by kengruven** [comment]
I can confirm this bug, with IE11 and MDL 1.1.1.  This document causes IE11 Print Preview to hang:

```
<html>
  <head>
    <link rel="stylesheet" href="https://code.getmdl.io/1.1.1/material.indigo-pink.min.css">
  </head>
  <body>
    hello, world
  </body>
</html>
```

It also happens with the non-minified `material.css` file.  I'll take a quick look and see if I can narrow it down any.

**Comment by kengruven** [comment]
The MDL `@media print` block is what's triggering this.  Here's a small stylesheet that hangs IE11 printing:

```
@media print {
  *:first-line {
    text-shadow: none !important;
  }
}
```

If I remove the `*:first-line` from the `@media print {}` block in material.css, it prints fine.

**Comment by sgomes** [comment]
Thanks for identifying the problem, @kengruven! I think it's safe to remove that media query altogether. @surma @Garbee What do you think?

**Comment by Garbee** [comment]
SGTM. If someone notices this removal then we can simply instruct them to add it in their code if they don't care about IE11 support.

**Comment by sgomes** [comment]
Fixed with #4169.

**Comment by TANGSTEN** [comment]
Hi Google, 

This issue was not fixed in IE11. Material Design Lite will break the IE11 print function. The file that breaks IE11 is material.css. The break is that only 1 page is printed any other attempt including this javascript work around will not fix it.

This is the JavaScript work around code everyone ravs about but wont fix the underlining issue in material.css:
var printContents = document.getElementById("YourDiv").innerHTML;
var popupWin = window.open('', '_blank', 'width=960,height=540');
popupWin.document.open();
popupWin.document.write('<html><head><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge" /><title>Indirect TPMS Calibration (Interactive)</title><link rel="stylesheet" href="css/material.css" type="text/css"></head><body onload="window.print();">' + printContents + '</body></html>');
popupWin.document.close();

The block of code in material.css that will break the print function is this:

/* Everything below here breaks IE11 Print Function */
.mdl-layout { width: 100%; height: 100%; display: flex; flex-direction: column; overflow-y: auto; overflow-x: hidden; position: relative; }
.mdl-layout:not(.is-small-screen) .mdl-layout--small-screen-only { display: none; }
.mdl-layout__container { position: absolute; width: 100%; height: 100%; }
.mdl-layout__drawer.is-visible ~ .mdl-layout__content.mdl-layout__content { overflow: hidden; }
.mdl-layout__header .mdl-layout__drawer-button { position: absolute; color: rgb(255, 255, 255); background-color: inherit; }
.mdl-layout__header { display: flex; flex-flow: column nowrap; justify-content: flex-start; box-sizing: border-box; flex-shrink: 0; width: 100%; margin: 0px; padding: 0px; border: none; min-height: 64px; max-height: 1000px; z-index: 3; background-color: rgb(63, 81, 181); color: rgb(255, 255, 255); box-shadow: rgba(0, 0, 0, 0.14) 0px 2px 2px 0px, rgba(0, 0, 0, 0.2) 0px 3px 1px -2px, rgba(0, 0, 0, 0.12) 0px 1px 5px 0px; transition-duration: 0.2s; transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1); transition-property: max-height, box-shadow; }
.mdl-layout--fixed-drawer.is-upgraded:not(.is-small-screen) > .mdl-layout__header { margin-left: 240px; width: calc(100% - 240px); }
.mdl-layout__header > .mdl-layout-icon { position: absolute; left: 40px; top: 16px; height: 32px; width: 32px; overflow: hidden; z-index: 3; display: block; }
.mdl-layout.has-drawer .mdl-layout__header > .mdl-layout-icon { display: none; }
.mdl-layout__header--waterfall.is-casting-shadow { box-shadow: rgba(0, 0, 0, 0.14) 0px 2px 2px 0px, rgba(0, 0, 0, 0.2) 0px 3px 1px -2px, rgba(0, 0, 0, 0.12) 0px 1px 5px 0px; }
.mdl-layout__header-row { display: flex; flex-flow: row nowrap; flex-shrink: 0; box-sizing: border-box; align-self: stretch; align-items: center; height: 64px; margin: 0px; padding: 0px 40px 0px 80px; }
.mdl-layout--no-drawer-button .mdl-layout__header-row { padding-left: 40px; }
.mdl-layout__header--scroll .mdl-layout__header-row { width: 100%; }
.mdl-layout__header-row .mdl-navigation { margin: 0px; padding: 0px; height: 64px; flex-direction: row; align-items: center; }
.mdl-layout__header-row .mdl-navigation__link { display: block; color: rgb(255, 255, 255); line-height: 64px; padding: 0px 24px; }
.mdl-layout__content { position: relative; display: inline-block; overflow-y: auto; overflow-x: hidden; flex-grow: 1; z-index: 1; }
/* Everything above here breaks IE11 Print Function */

Removing the last line in that block will cause the print function to spit out 300 pages. 
Please help me resolve this I SPENT A LOT OF TIME ISOLATING THIS CODE.

Thanks - Joe
